packages/loader/container-loader/src/container.ts

```diff
@@ -1125,7 +1125,8 @@ export class Container
 			savedOps: this.savedOps,
 			url: this.resolvedUrl.url,
 			term: OnlyValidTermValue,
-			clientId: this.clientId,
+			// no need to save this if there is no pending runtime state
+			clientId: pendingRuntimeState !== undefined ? this.clientId : undefined,
```

Comment:
This is necessary for this scenario:
- closeAndGetPendingLocalState() is called with no pending changes, so sequenced local ops are not saved by runtime
- Then, a container is hydrated with this blob and generates some pending local changes
- closeAndGetPendingLocalState() is called again on this container. Since it was rehydrated with the clientId from the stash blob, runtime saves this clientId
- When this stash blob is used to rehydrate another container, it treats replayed sequenced ops from the first container as local, even though they weren't saved because the previous container didn't know about them.

This is avoided by Container not saving/rehydrating the clientId when there are no pending changes, in which case it is unnecessary.

We could also just not save the clientId in runtime when stashing if there are pending ops but no saved ops, but that would be somewhat awkward, and I think this is how Container "should" behave anyway.

Reply:
would this comment make sense to put in the code as a comment?